### PR TITLE
[FLINK-22655]When using -i <init.sql> option to initialize SQL Client session It should be possible to annotate the script with --

### PR DESCRIPTION
**Pull request by JasonLeeCoding**
## What is the purpose of the change

*this change is in order to use comments in the initialization script with '--' in the SQL Client*

## Brief change log

*add replaceAll in CliStatementSplitter#splitContent*

## Verifying this change

This change added tests and can be verified as follows:

*Modify the original test file CliStatementSplitterTest.java*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by JasonLeeCoding** [comment]
I got an error java.lang.AssertionError  testCancelExecutionInNonInteractiveMode throws an exception at line 303 of the code
Can anyone help me take a look ?

**Comment by fsk119** [comment]
Thanks for your contribution. Please use rebase rather than merge in FLINK. What's more, please also add tests in `CliStatementSplitterTest` and `SqlClientTest` to verify the code is right.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java [hunk 1]
```diff
@@ -49,7 +50,10 @@ public static boolean isStatementComplete(String statement) {
         for (String line : content.split("\n")) {
             if (isEndOfStatement(line)) {
                 buffer.add(line);
-                statements.add(String.join("\n", buffer));
+                statements.add(
+                        buffer.stream()
```
Comment (fsk119) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java:54]: Please also take a look at line 63.

**Comment by fsk119** [comment]
> I got an error java.lang.AssertionError testCancelExecutionInNonInteractiveMode throws an exception at line 303 of the code
> Can anyone help me take a look ?

Please take a look at the logic in the test.. We compare the received statement with input statement in the test. However, we clear all the comments in `CliStatementSplitter`, which cause the test fails.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java (hunk 1 above)
Comment (fsk119) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java:54]: Could you filter the empty line after `map` operation, e.g.

```
buffer.stream()
  .map(...)
  .filter(...) // only keep the non-empty line here
  .collect(...)
```

Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java:54]: Thanks , I will fix these problems

**Comment by fsk119** [comment]
Please read doc about code formatting in https://ci.apache.org/projects/flink/flink-docs-master/docs/flinkdev/ide_setup/#code-formatting

**Comment by JasonLeeCoding** [comment]
OK I submitted a new PR  https://github.com/apache/flink/pull/15980  and I will close this PR

**Pull request closed without merging**